library/src/test/java/org/apache/apex/malhar/lib/window/impl/WindowedJoinOperatorWatermarkTest.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.window.impl;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.apache.apex.malhar.lib.window.WindowState;
+
+/**
+ *
+ */
+public class WindowedJoinOperatorWatermarkTest
+{
+  @Test
+  public void WindowedJoinOperatorWatermarkTest()
+  {
+    WindowedJoinOperatorImpl op = new WindowedJoinOperatorImpl();
+    op.setWindowStateStorage(new InMemoryWindowedStorage<WindowState>());
+    op.controlInput.process(new WatermarkImpl(1000000));
+    op.endWindow();
+    Assert.assertEquals(1000000, op.currentWatermark);
+    op.controlInput2.process(new WatermarkImpl(200000));
+    op.endWindow();
+    Assert.assertEquals(200000, op.currentWatermark);
+    op.controlInput2.process(new WatermarkImpl(20000));
+    op.endWindow();
```

Comment:
The min of the latest watermarks from input streams is the current watermark of the join operator, and cannot be smaller than the previous watermark of the join operator, but it can be equal, and in which case, you do not generate a watermark. 

Watermark tuple stream from one source (whether from one of the input ports or the join operator itself) must have a timestamp that is ever increasing.